### ISSUE TITLE
No clusters survive Monte Carlo cluster-level FWE-correction in 0.0.4

### ISSUE THREAD

**Issue by alexenge** [neurostuff/NiMARE#429]
First of all, a massive thank you to everyone involved in NiMARE. It's really exciting to see so many different approaches to CBMA and IBMA implemented in a user-friendly manner in Python! Also, this is one of my first issues raised on GH, so please excuse me being imprecise (and/or stupid).

I've noticed that after upgrading to version 0.0.4, an ALE analysis that yielded really robust results after FWE correction (both in GingerALE and in NiMARE Version 0.0.3) suddenly returns empty `'z_level-cluster_corr-FWE_method-montecarlo'`-maps  (whereas the unthresholded ALE-map and the unthresholded and voxel-level corrected *z*-maps look the same as before).

As an example, I'm attaching the the cluster-level corrected *z*-maps produced by the following code, run with the two latest versions of NiMARE:

```python
def run_ale(text_file, voxel_thresh, n_iters, output_dir):
    from nimare import io, meta, correct
    dset = io.convert_sleuth_to_dataset(text_file=text_file)
    ale = meta.cbma.ALE()
    res = ale.fit(dset)
    corr = correct.FWECorrector(method='montecarlo', voxel_thresh=voxel_thresh, n_iters=n_iters)
    cres = corr.transform(result=res)
    cres.save_maps(output_dir=output_dir)


run_ale(text_file='foci.txt', voxel_thresh=0.001, n_iters=1000, output_dir='.')
```

[foci.txt](https://github.com/neurostuff/NiMARE/files/5757394/foci.txt)
[v003_z_level-cluster_corr-FWE_method-montecarlo.nii.gz](https://github.com/neurostuff/NiMARE/files/5757397/v003_z_level-cluster_corr-FWE_method-montecarlo.nii.gz)
[v003_plot_z.png](https://user-images.githubusercontent.com/66270814/103413072-576f4200-4b78-11eb-8d7d-708f402b518c.png)
[v003_plot_z_cluster_corrected.png](https://user-images.githubusercontent.com/66270814/103413069-55a57e80-4b78-11eb-9e1d-5a2fce5d7e6e.png)
[v004_z_level-cluster_corr-FWE_method-montecarlo.nii.gz](https://github.com/neurostuff/NiMARE/files/5757392/v004_z_level-cluster_corr-FWE_method-montecarlo.nii.gz)
[v004_plot_z.png](https://user-images.githubusercontent.com/66270814/103413125-82f22c80-4b78-11eb-9df8-0b73cb43ba6b.png)
[v004_plot_z_cluster_corrected.png](https://user-images.githubusercontent.com/66270814/103413128-84235980-4b78-11eb-996d-14c43ac0b76b.png)

I think the problem is also visible in your example at [https://nimare.readthedocs.io/en/latest/auto_examples/02_meta-analyses/plot_cbma.html#ale](https://nimare.readthedocs.io/en/latest/auto_examples/02_meta-analyses/plot_cbma.html#ale).

It would be great if you could look into this to check if it's my fault or if something is wrong with the code. Thanks in advance for all your efforts and have a good start into 2021!

**Comment by tsalo** [comment]
Thank you for opening this issue. This does, in fact, appear to be a bug; one that _I_ introduced recently in #409. Specifically, in #409 I changed how the null distribution is stored (from a true null distribution to a set of histogram weights), but forgot to update a specific method that uses that distribution (`CBMAEstimator._p_to_summarystat()`), so it was treating raw histogram weights like a null distribution. As a result, the "maximum cluster size" from each Monte Carlo permutation was essentially the whole brain, and no _real_ clusters were going to be significant based on that.

After correcting the issue, the cFWE-corrected z-map looks much more reasonable:

<img width="555" alt="Screen Shot 2020-12-31 at 10 53 24 AM" src="https://user-images.githubusercontent.com/8228902/103417024-0ef35c80-4b57-11eb-85fc-4c24dac5fa32.png">

I am going to open a PR to deal with this posthaste.

**Comment by tsalo** [comment]
@alexenge, before I make a new release, would you mind running your analysis with the current `master` branch to see if you get what you were expecting?

**Comment by alexenge** [comment]
Thanks for the quick reply! The version on the `master` branch works like charm and reproduces the results from the released Version 0.0.3.

**Comment by tsalo** [comment]
That's good to hear. I will make the new release then.
